### PR TITLE
Config error messaging

### PR DESCRIPTION
**Pull request by tisAliG**
Issue #39

### CODE REVIEW

File: generators/pattern-library/index.js [hunk 1]
```diff
@@ -14,12 +14,17 @@ module.exports = init;
...
 
-    if ( Array.isArray( config ) ) {
+    // If we have errors in the array, reject and handoff those errors to the user
+    if ( config.errors.length ) {
+
+        return new Promise( ( resolve, reject ) => {
 
-        return new Promise( function ( resolve, reject ) {
-            return reject( config );
+            return reject( config.errors );
```
Comment (vbalcom) [inline generators/pattern-library/index.js]: Should we be returning only the errors array? Or the whole config which has an errors key?

File: test/configure.js [hunk 2]
```diff
@@ -29,7 +29,13 @@ suite( 'Configure: Settings', () => {
         }, data => {
 
             assert.isArray( data, 'Sugarcoat returns an array when we are missing a dest.' );
-            assert.propertyVal( data[0], 'key', 'settings.dest', 'We have dest in the error array');
+
+            data.forEach( ( errorObj, index ) => {
+
+                assert.instanceOf( errorObj, Error, 'The object was an Error Object.' );
```
Comment (vbalcom) [inline test/configure.js]: Looks like we're now testing that there is an Error object. But we don't know which error was returned. Can we make this more specific to the test?

File: generators/pattern-library/index.js (hunk 1 above)
Comment (tisAliG) [inline generators/pattern-library/index.js]: When there is an error, the user won't be using anything else other than the errors, so there isn't a reason I can see why they would need the whole config.

File: test/configure.js (hunk 2 above)
Comment (tisAliG) [inline test/configure.js]: boom. Moved the errors into a separate file. That way we have them all in one place, and we can use that for testing as well.

**Pull request merged**